mifosng-provider/src/integrationTest/java/org/mifosplatform/integrationtests/SearchTest.java

```diff
@@ -0,0 +1,230 @@
+import org.mifosplatform.integrationtests.common.GroupHelper;
+import org.mifosplatform.integrationtests.common.loans.LoanTransactionHelper;
+import org.mifosplatform.integrationtests.common.savings.SavingsAccountHelper;
+import org.mifosplatform.integrationtests.common.Utils;
+import org.mifosplatform.integrationtests.LoanApplicationApprovalTest;
+
+import com.jayway.restassured.builder.RequestSpecBuilder;
+import com.jayway.restassured.builder.ResponseSpecBuilder;
+import com.jayway.restassured.http.ContentType;
+import com.jayway.restassured.specification.RequestSpecification;
+import com.jayway.restassured.specification.ResponseSpecification;
+
+public class SearchTest {
+	 private ResponseSpecification responseSpec;
+	 private RequestSpecification requestSpec;
+	 private LoanTransactionHelper loanTransactionHelper;
+	   
+	   
+	@Before   
+	public void setup() {
+	        Utils.initializeRESTAssured();
+	        this.requestSpec = new RequestSpecBuilder().setContentType(ContentType.JSON).build();
+	        this.requestSpec.header("Authorization", "Basic " + Utils.loginIntoServerAndGetBase64EncodedAuthenticationKey());
+	        this.responseSpec = new ResponseSpecBuilder().expectStatusCode(200).build();
+	        this.loanTransactionHelper = new LoanTransactionHelper(this.requestSpec, this.responseSpec);
+            
+	    }
+
+
+	@Test
+	public void Test() {			 
+		
+		 final Integer clientID= ClientHelper.createClient(this.requestSpec, this.responseSpec);
+		 final Integer groupID = GroupHelper.createGroup(this.requestSpec, this.responseSpec);
+		 final Integer centerId= CenterHelper.createCenter(this.requestSpec, this.responseSpec);
+		 final Integer savingsAccountId=SavingsAccountHelper.openSavingsAccount(this.requestSpec, this.responseSpec,clientID,"10000");	     
+		 String  externalId=null;
+		 String  accountNo=null;
+		 String  displayName=null; 		 
+		 String likeMatch_externalId=null;
+		 String likeMatch_accountNo=null;
+		 String likeMatch_displayName=null;
+		 String likeMatch_clientId=null;
+		 String likeMatch_groupId=null;
+		 
+         ArrayList<HashMap> clientdetail=new ArrayList<HashMap>();
+	     clientdetail =SearchHelper.likeMatchSearchClient(requestSpec, responseSpec, String.valueOf(clientID));   
+	        if (clientdetail != null) {
+	        for (HashMap oneResult : clientdetail) {	    		
+	        externalId =(String)oneResult.get("entityExternalId");
+	        accountNo  =(String)oneResult.get("entityAccountNo");	
+	        displayName=(String)oneResult.get("entityName");	        	        
+	        likeMatch_externalId=externalId.substring(1,3);
+	        likeMatch_accountNo= accountNo.substring(1,3);
+	        likeMatch_displayName=displayName.substring(1, 3);
+	        likeMatch_clientId=String.valueOf(clientID).substring(1);
+	    	}
+	    	 
+	         
+	    	 ArrayList<HashMap> exactMatchClientDetail_Search_By_ExternalId=SearchHelper.exactMatchSearchClient(requestSpec, responseSpec, externalId);
```

Comment:
instead of calling exactMatchSearchClient to validate each contraint, get the object and use get("accountNo").similarly for others